core/src/main/scala/fs2/async/mutable/Semaphore.scala

```diff
@@ -56,7 +56,8 @@ object Semaphore {
 
   /** Create a new `Semaphore`, initialized with `n` available permits. */
   def apply[F[_]](n: Long)(implicit F: Async[F]): F[Semaphore[F]] = {
-    def ensureNonneg(n: Long) = if (n < 0) throw new IllegalArgumentException("n must be nonnegative, was: " + n)
+    def ensureNonneg(n: Long) = assert(n >= 0, s"n must be nonnegative, was: $n ")
```

Comment:
ok, I see your point. I'll introduce something like this in util?. I just think the if and throw is sort of unexpected. Maybe checkArgument(cond:Boolean, msg:=> String):Unit?